aws/resource_aws_guardduty_filter.go

```diff
@@ -0,0 +1,296 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	//"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGuardDutyFilter() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGuardDutyFilterCreate,
+		Read:   resourceAwsGuardDutyFilterRead,
+		Update: resourceAwsGuardDutyFilterUpdate,
+		Delete: resourceAwsGuardDutyFilterDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"auto_archive": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"rank": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"filters": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
These elements should match the non-deprecated values of the `guardduty.Condition` struct. You can make all of the attributes `Optional`, and then combinations of `ConflictsWith`, `ExactlyOneOf`, and `AtLeastOneOf` as applicable for the filter combinations.